lib/factory_bot/definition.rb

```diff
@@ -173,7 +173,12 @@ def expand_enum_traits(klass)
     end
 
     def automatically_register_defined_enums(klass)
-      klass.defined_enums.each_key { |name| register_enum(Enum.new(name)) }
+      klass.defined_enums.each_key do |name|
+        next if registered_enums.any? do |registered_enum|
```

Comment:
This might be slightly easier, and have better performance when there are many enums, if we change registered enums to a hash of name -> Enum. Then checking whether it was registered would be `registered_enums.has_key?(name)`

Reply:
I agree with your comment, this nested loop could loop through all enums every time.

Reply:
updated 👍